### PR TITLE
Change includeArchived to archivedStatus param

### LINKED ISSUE #6062: [Unarchive patient] Patients query updates
### Background
To implement the unarchive patient flow we need to be able to retrieve `archived patients` only . For a more detailed explanation of the unarchive patient flow please check the findings of the [spike ticket](https://github.com/CDCgov/prime-simplereport/issues/5708).

### Action requested
Modify the `patients query` to receive a parameter that allows to refine the patients search to three states: `all` (all patients) | `archived` (archive patients only) | `unarchived` (unarchived patients only).
Remove the `includeArchived` parameter and make necessary changes to migrate the logic that uses it to the new parameter.

### Acceptance Criteria
- Patients query is able to return only archived patients, only unarchived patients and all type of patients depending on the parameter value passed.
- The application's functionality is not impacted in any way.
- The usage of the `includeArchive` parameter has been cleaned up.

### Additional Context
This query allows to refine the retrieval of patients somehow with the includeArchive parameter however this boolean parameter does not allow for a selective retrieval of patients.

### PR DESCRIPTION
**Pull request by emyl3**
# BACKEND & FRONTEND PULL REQUEST

## Related Issue

Resolves #6062 

## Changes Proposed
Update `getPatients` and `patientsCount` to accept a parameter that can return deleted users, undeleted users, or all users.

## Additional Information
Test the new `archivedStatus` parameter!
Update the values below as you see fit.

patients
```
patients(
      facilityId: $facilityId
      pageNumber: 0
      pageSize: 100
      archivedStatus: $archivedStatus
      namePrefixMatch: $namePrefixMatch
      includeArchivedFacilities: $includeArchivedFacilities
    ) {
      internalId
      firstName
      lastName
      middleName
      birthDate
      gender
      telephone
      email
      emails
      phoneNumbers {
        type
        number
      }
    }
```

patientsCount
```
    patientsCount(
      facilityId: $facilityId
      archivedStatus: $archivedStatus
      namePrefixMatch: $namePrefixMatch
    ) 
```

## Testing
Deployed on dev4 for testing

### CODE REVIEW

File: frontend/src/app/testResults/TestResultsList.tsx [hunk 1]
```diff
@@ -452,11 +466,10 @@ export const DetachedTestResultsList = ({
                   setStartDateError("");
                   setEndDateError("");
                 }}
-                disabled={
```
Comment (emyl3) [inline frontend/src/app/testResults/TestResultsList.tsx:455]: ⚠️ updated this to address a code smell related to code complexity being over 15

**Comment by fzhao99** [comment]
NB for reviewers: "b2daabe8-11c8-4006-aa5f-0163869d369b" is a facility Id you can used with archived (5 total) / unarchived (4 total) patients

File: backend/src/test/java/gov/cdc/usds/simplereport/service/PersonServiceTest.java [hunk 2]
```diff
@@ -145,12 +146,18 @@ void roundTrip() {
         "French",
         null);
     List<Person> all =
```
[the remark this answers is not in the document]
Comment (emyl3) [reply backend/src/test/java/gov/cdc/usds/simplereport/service/PersonServiceTest.java:148]: Yes! Nice catch! I forgot to push up that change 😶‍🌫️ Will tag you in some comments on this PR 😅

File: backend/src/test/java/gov/cdc/usds/simplereport/service/PersonServiceTest.java [hunk 3]
```diff
@@ -654,38 +691,48 @@ void getPatients_search_OrgAdminUser() {
...
+        _service.getPatients(site2Id, 0, 5, ArchivedStatus.UNARCHIVED, null, false).get(0);
     _service.setIsDeleted(frank.getInternalId(), true);
 
     // all facilities, not deleted, "ma"
-    List<Person> patients = _service.getPatients(null, 0, 100, false, "ma", false);
+    List<Person> patients =
+        _service.getPatients(null, 0, 100, ArchivedStatus.UNARCHIVED, "ma", false);
     assertPatientList(patients, GALE, JANNELLE, KACEY, ELIZABETH, HEINRICK);
+    // all facilities, deleted
+    List<Person> deletedPatients =
```
Comment (emyl3) [inline backend/src/test/java/gov/cdc/usds/simplereport/service/PersonServiceTest.java:707]: @fzhao99 -- testing patients for "ARCHIVED" status here, all facilities

File: backend/src/test/java/gov/cdc/usds/simplereport/service/PersonServiceTest.java [hunk 4]
```diff
@@ -654,38 +691,48 @@ void getPatients_search_OrgAdminUser() {
...
+        _service.getPatients(null, 0, 100, ArchivedStatus.ARCHIVED, null, false);
+    assertPatientList(deletedPatients, CHARLES, FRANK);
 
     // site2, not deleted, "ma"
-    patients = _service.getPatients(site2Id, 0, 100, false, "ma", false);
+    patients = _service.getPatients(site2Id, 0, 100, ArchivedStatus.UNARCHIVED, "ma", false);
     assertPatientList(patients, JANNELLE, KACEY);
+    // site2, deleted, "fran"
+    deletedPatients = _service.getPatients(site2Id, 0, 100, ArchivedStatus.ARCHIVED, "fran", false);
```
Comment (emyl3) [inline backend/src/test/java/gov/cdc/usds/simplereport/service/PersonServiceTest.java:715]: @fzhao99 -- testing patients for "ARCHIVED" status here from a specific facility with text "fran"

File: backend/src/test/java/gov/cdc/usds/simplereport/service/PersonServiceTest.java [hunk 5]
```diff
@@ -714,34 +764,44 @@ void getPatients_counts() {
...
     // Delete Frank (_site2)
     _service.setIsDeleted(patients_site2.get(0).getInternalId(), true);
 
-    assertEquals(10, _service.getPatientsCount(null, false, null, false));
-    assertEquals(12, _service.getPatientsCount(null, true, null, false));
-    assertEquals(6, _service.getPatientsCount(site2Id, false, null, false));
-    assertEquals(7, _service.getPatientsCount(site2Id, true, null, false));
+    assertEquals(10, _service.getPatientsCount(null, ArchivedStatus.UNARCHIVED, null, false));
+    assertEquals(12, _service.getPatientsCount(null, ArchivedStatus.ALL, null, false));
+    assertEquals(2, _service.getPatientsCount(null, ArchivedStatus.ARCHIVED, null, false));
```
Comment (emyl3) [inline backend/src/test/java/gov/cdc/usds/simplereport/service/PersonServiceTest.java:787]: @fzhao99 -- testing patients count for "ARCHIVED" status here, all facilities

File: backend/src/test/java/gov/cdc/usds/simplereport/service/PersonServiceTest.java [hunk 6]
```diff
@@ -714,34 +764,44 @@ void getPatients_counts() {
...
+    assertEquals(5, _service.getPatientsCount(null, ArchivedStatus.UNARCHIVED, "ma", false));
+    assertEquals(2, _service.getPatientsCount(site2Id, ArchivedStatus.UNARCHIVED, "ma", false));
+    assertEquals(4, _service.getPatientsCount(site1Id, ArchivedStatus.ALL, "ma", false));
+    assertEquals(4, _service.getPatientsCount(null, ArchivedStatus.UNARCHIVED, "mar", false));
+    assertEquals(2, _service.getPatientsCount(null, ArchivedStatus.UNARCHIVED, "MARTHA", false));
+
+    assertEquals(0, _service.getPatientsCount(null, ArchivedStatus.UNARCHIVED, "M", false));
+    assertEquals(0, _service.getPatientsCount(null, ArchivedStatus.UNARCHIVED, "", false));
+
+    assertEquals(1, _service.getPatientsCount(null, ArchivedStatus.ARCHIVED, "fra", false));
```
Comment (emyl3) [inline backend/src/test/java/gov/cdc/usds/simplereport/service/PersonServiceTest.java:802]: @fzhao99 -- testing patient count for "ARCHIVED" status here, all facilities or particular facility with text "fra" or "charles"

**Comment by emyl3** [comment]
Thank you @fzhao99 for flagging that. Let me know what you think! 😸

**Review by fzhao99 — APPROVED** [review]
lgtm!

**Review by johanna-skylight — COMMENTED** [review]
Code looks awesome overall. The only question that I have is whether or not this will be backwards compatible with the UI (as the UI changes are going in the same PR).

How will the two graphql operations behave for requests that are still relying on the old parameter? (case when the UI is running old code)

**Comment by emyl3** [comment]
> Code looks awesome overall. The only question that I have is whether or not this will be backwards compatible with the UI (as the UI changes are going in the same PR).
> 
> How will the two graphql operations behave for requests that are still relying on the old parameter? (case when the UI is running old code)

OOO that is a very good point. Thank you for bringing this up. This change will definitely break some views for the X number of mins a user has the old frontend. 

I will put up two separate PRs and probably close this one:
- one PR to introduce and leverage this new parameter (`archivedStatus`)
- another PR to deprecate and remove support for the old parameter (`includeArchived`)

Let me know if this sounds good to you, @johanna-skylight! Thank you! 😸

**Comment by emyl3** [comment]
closed in favor of https://github.com/CDCgov/prime-simplereport/pull/6156

**Pull request closed without merging**